app/scripts/modules/core/src/account/AccountSelectField.tsx

```diff
@@ -1,13 +1,142 @@
+import * as React from 'react';
+import { $q } from 'ngimport';
+import { flatten, has, intersection, isEqual, map, uniq, xor } from 'lodash';
+
 import { IAccount } from 'core/account';
+import { AccountService } from 'core/account/AccountService';
 
 export interface IAccountSelectFieldProps {
   accounts: IAccount[] | string[];
-  component: Object;
+  component: { [key: string]: any };
   field: string;
   provider: string;
   loading?: boolean;
   onChange?: (account: string) => void;
   labelColumns?: number;
   readOnly?: boolean;
-  multiselect?: boolean;
+}
+
+export interface IAccountSelectFieldState {
+  accountContainsExpression: boolean;
+  mergedAccounts: string[];
+  primaryAccounts: string[];
+  secondaryAccounts: string[];
+}
+
+const isExpression = (account: string) => !!account && account.includes('${');
+
+export class AccountSelectField extends React.Component<IAccountSelectFieldProps, IAccountSelectFieldState> {
+  public state: IAccountSelectFieldState = {
+    accountContainsExpression: false,
+    mergedAccounts: [],
+    primaryAccounts: [],
+    secondaryAccounts: [],
+  };
+
+  private groupAccounts = (accounts: IAccount[] | string[]) => {
+    const { component, field, onChange, provider } = this.props;
+
+    if (!accounts || !accounts.length) {
+      return;
+    }
+    if (has(component, field) && isExpression(component[field])) {
+      this.setState({ accountContainsExpression: true });
+      return;
+    }
+
+    const accountsAreObjects = Boolean((accounts[0] as IAccount).name);
+    let getAccountDetails = provider ? AccountService.getAllAccountDetailsForProvider(provider) : $q.when([]);
+
+    if (provider && accountsAreObjects) {
+      const providers = uniq(map(accounts as IAccount[], 'type'));
+      getAccountDetails = $q
+        .all(providers.map(p => AccountService.getAllAccountDetailsForProvider(p)))
+        .then(details => flatten(details));
+    }
+
+    getAccountDetails.then(details => {
+      const accountNames: string[] = accountsAreObjects ? map(accounts as IAccount[], 'name') : (accounts as any);
+      let mergedAccounts = accountNames;
+      let primaryAccounts: string[] = [];
+      let secondaryAccounts: string[] = [];
+
+      if (accountNames) {
+        primaryAccounts = accountNames.sort();
+      }
+      if (accountNames && accountNames.length && details.length) {
+        primaryAccounts = accountNames
+          .filter(account => {
+            return details.some(detail => detail.name === account && detail.primaryAccount);
+          })
+          .sort();
+        secondaryAccounts = xor(accountNames, primaryAccounts).sort();
+        mergedAccounts = flatten([primaryAccounts, secondaryAccounts]);
+      }
+
+      if (component) {
+        mergedAccounts = mergedAccounts || [];
+        const comp = flatten([component[field]]);
```

Comment:
this seems off since `flatten` is a shallow operation and you're passing in an array of one element.  I think this will always be equivalent to `const comp = component[field]`

Reply:
Good catch. Leftover cruft from not copying over the multi select property from the angular directive since we aren't using it anywhere.